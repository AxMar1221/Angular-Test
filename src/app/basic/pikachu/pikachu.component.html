<app-navbar></app-navbar>
<div class="container mt-5">
  <div *ngIf="!pikachu, else elseBlock">
    <h1>Pika Pika</h1>
    <h2>Loading...</h2>
  </div>

  <ng-template #elseBlock>
    <h3>{{ pikachu?.name | titlecase }}</h3>
    <hr>
    <img [src]="pikachu?.sprites?.front_default" [alt]="pikachu?.name">
  </ng-template>
</div>
