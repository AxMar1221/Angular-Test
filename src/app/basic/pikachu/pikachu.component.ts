import { Component, OnInit } from '@angular/core';
import { Pokemon } from '../interfaces';
import { PikachuService } from '../services/pikachu.service';

@Component({
  selector: 'app-pikachu',
  templateUrl: './pikachu.component.html',
})
export class PikachuComponent implements OnInit {

  public pikachu?: Pokemon;

  constructor( private pokemonService: PikachuService ) { }

  ngOnInit(): void {
    this.pokemonService.getPokemon(25)
    .subscribe( pokemon => {
      this.pikachu = pokemon;
      // console.log( pokemon );
    });
  }

}
