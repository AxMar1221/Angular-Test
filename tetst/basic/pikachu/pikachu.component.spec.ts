import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';

import { PikachuComponent } from '../../../src/app/basic/pikachu/pikachu.component';
import { PikachuService } from '../../../src/app/basic/services/pikachu.service';

describe('PikachuComponent', () => {
  let component: PikachuComponent;
  let fixture: ComponentFixture<PikachuComponent>;
  let compile: HTMLElement;
  let service: PikachuService;
  let httpMock: HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ PikachuComponent ],
      imports: [ HttpClientTestingModule ],
      providers: [ PikachuService]
    })
    .compileComponents();

    fixture = TestBed.createComponent(PikachuComponent);
    component = fixture.componentInstance;
    service = TestBed.inject( PikachuService );
    httpMock = TestBed.inject( HttpTestingController );

    fixture.detectChanges();
    compile = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe de hacer macht con el snapshot', () => {
    expect(compile.innerHTML).toMatchSnapshot();
  });

  test('debeb de mostrar un loading al inicio', () => {
    const h2 = compile.querySelector('h2');
    expect( h2?.textContent ).toContain('Loading...')
  });

  test('debe de cargar Pikachu inmediatamente', ()=> {

    const dummyPokemon = {
      name: 'Pika Pika!',
      sprites: {
        front_default: 'https://pikachu.com/spritre.com'
      }
    };

    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/25');
    expect( request.request.method ).toBe('GET');
    request.flush( dummyPokemon );

    fixture.detectChanges();
    // console.log(compile.innerHTML);
    const h3 = compile.querySelector('h3');
    const img = compile.querySelector('img');

    expect( h3?.textContent?.toLowerCase() ).toContain( dummyPokemon.name.toLowerCase() );
    expect( img?.src ).toBe( dummyPokemon.sprites.front_default );
    expect( img?.alt ).toBe( dummyPokemon.name );

  })
});
