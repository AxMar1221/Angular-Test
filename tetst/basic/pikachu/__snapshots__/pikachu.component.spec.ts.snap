// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PikachuComponent debe de hacer macht con el snapshot 1`] = `
"<app-navbar></app-navbar><div class=\\"container mt-5\\"><div><h1>Pika Pika</h1><h2>Loading...</h2></div><!--bindings={
  \\"ng-reflect-ng-if-else\\": \\"[object Object]\\"
}--><!--container--></div>"
`;
